packages/grafana-ui/src/components/Table/FilterPopup.tsx

```diff
@@ -0,0 +1,159 @@
+  const uniqueValues = useMemo(() => calculateUniqueFieldValues(preFilteredRows, field), [preFilteredRows, field]);
+  const options = useMemo(() => valuesToOptions(uniqueValues), [uniqueValues]);
+  const filteredOptions = useMemo(() => getFilteredOptions(options, filterValue), [options, filterValue]);
+  const [values, setValues] = useState<SelectableValue[]>(filteredOptions ?? []);
+
+  const onCancel = useCallback((event?: React.MouseEvent) => onClose(), [onClose]);
+
+  const onFilter = useCallback(
+    (event: React.MouseEvent) => {
+      const filtered = values.length ? values : undefined;
+
+      setFilter(filtered);
+      onClose();
+    },
+    [setFilter, values, onClose]
+  );
+
+  const onClearFilter = useCallback(
+    (event: React.MouseEvent) => {
+      setFilter(undefined);
+      onClose();
+    },
+    [setFilter, onClose]
+  );
+
+  const clearFilterVisible = useMemo(() => filterValue !== undefined, [filterValue]);
+
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  return (
+    <ClickOutsideWrapper onClick={onCancel} useCapture={true}>
+      <div className={cx(styles.filterContainer)} onClick={stopPropagation}>
+        <VerticalGroup spacing="lg">
+          <VerticalGroup spacing="xs">
+            <Label>Filter by values:</Label>
+            <div className={cx(styles.listDivider)} />
+            <FilterList onChange={setValues} values={values} options={options} />
+          </VerticalGroup>
+          <HorizontalGroup spacing="lg">
+            <HorizontalGroup>
+              <Button size="sm" onClick={onFilter}>
+                Ok
+              </Button>
+              <Button size="sm" variant="secondary" onClick={onCancel}>
+                Cancel
+              </Button>
+            </HorizontalGroup>
+            {clearFilterVisible && (
+              <HorizontalGroup>
+                <Button variant="link" size="sm" onClick={onClearFilter}>
+                  Clear filter
+                </Button>
+              </HorizontalGroup>
+            )}
+          </HorizontalGroup>
+        </VerticalGroup>
+      </div>
+    </ClickOutsideWrapper>
+  );
+};
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  filterContainer: css`
+    label: filterContainer;
+    width: 100%;
+    min-width: 250px;
+    height: 100%;
+    max-height: 400px;
+    background-color: ${theme.colors.bg1};
+    border: ${theme.border.width.sm} solid ${theme.colors.border2};
+    padding: ${theme.spacing.md};
+    margin: ${theme.spacing.sm} 0;
+    box-shadow: 0px 0px 20px ${theme.colors.dropdownShadow};
+    border-radius: ${theme.spacing.xs};
+  `,
+  listDivider: css`
+    label: listDivider;
+    width: 100%;
+    border-top: ${theme.border.width.sm} solid ${theme.colors.border2};
+    padding: ${theme.spacing.xs} ${theme.spacing.md};
+  `,
+}));
+
+const calculateUniqueFieldValues = (rows: any[], field?: Field) => {
```

Comment:
This functions seems like a candidate for a field util in grafana-data, WDYT?

Reply:
Hmm, not sure as it takes `rows: any[]` from the Table as an argument. If this was a pure field operation, I'd agree. I'll keep it here for now, but add tests as @leventebalogh suggested.